### PR TITLE
[FB Internal] Change Build Env to gcc-5

### PR DESCRIPTION
**Pull request by siying**
Summary: Default to build using gcc-5. Only apply to Facebook-only environments.

Test Plan:Run all existing tests. Make release

### CODE REVIEW

File: build_tools/fbcode_config.sh [hunk 1]
```diff
@@ -132,9 +132,10 @@ CFLAGS+=" -DROCKSDB_PLATFORM_POSIX -DROCKSDB_LIB_IO_POSIX -DROCKSDB_FALLOCATE_PR
 CXXFLAGS+=" $CFLAGS"
 
 EXEC_LDFLAGS=" $SNAPPY_LIBS $ZLIB_LIBS $BZIP_LIBS $LZ4_LIBS $ZSTD_LIBS $GFLAGS_LIBS $NUMA_LIB $TBB_LIBS"
-EXEC_LDFLAGS+=" -Wl,--dynamic-linker,/usr/local/fbcode/gcc-4.9-glibc-2.20/lib/ld.so"
+EXEC_LDFLAGS+=" -B$BINUTILS/gold"
```
Comment (siying) [inline build_tools/fbcode_config.sh]: Here is one thing I'm not sure. If I don't add this. I'll see a link error. After adding it, it seems to work and tests are passing and I can run the tools.

**Comment by IslamAbdelRahman** [comment]
Looks good to me but this don't affect buck, we need to fix it for buck

**Comment by siying** [comment]
@IslamAbdelRahman what we should do to change buck too other than changing PLATFORM file?

**Comment by siying** [comment]
For some reason I don't understand. utilities/spatialdb/spatial_db_test.cc is failing because in one test we are thinking moving the last 3 characters of an encoded string will generate a corrupted string, but it didn't. I changed it to 4 instead.

**Pull request closed without merging**